storage/measure.go

```diff
@@ -0,0 +1,221 @@
+package storage
+
+import (
+	"fmt"
+	"github.com/nknorg/tuna/util"
+	"log"
+	"math"
+	"net"
+	"path/filepath"
+	"sync"
+	"time"
+)
+
+const (
+	maxFavoriteLength = 10
+	maskSize          = 16
+	favoriteExpired   = 14 * 24 * time.Hour
+	badExpired        = 7 * 24 * time.Hour
+
+	FavoriteFileName = "favorite-node.json"
+	BadFileName      = "bad-node.json"
+)
+
+type FavoriteNode struct {
+	IP           string  `json:"ip"`
+	Address      string  `json:"address"`
+	Metadata     string  `json:"metadata"`
+	Delay        float32 `json:"delay"`
+	MinBandwidth float32 `json:"minBandwidth"`
+	MaxBandwidth float32 `json:"maxBandwidth"`
+	ExpiresAt    int64   `json:"expiredAt"`
+}
+
+type BadNodes = map[string]*BadNode
+
+type BadNode struct {
+	IP        string `json:"ip"`
+	MaskSize  int32  `json:"maskSize"`
+	Address   string `json:"address"`
+	ExpiresAt int64  `json:"expiredAt"`
+}
+
+type MeasureStorage struct {
+	path             string
+	favoriteFilePath string
+	badFilePath      string
+
+	FavoriteNodes *Storage
+
+	badNodeMutex sync.RWMutex
+	BadNodes     map[string]BadNodes
+
+	badNodeFileMutex      sync.RWMutex
+	favoriteNodeFileMutex sync.RWMutex
+}
+
+func NewMeasureStorage(path string) *MeasureStorage {
+	return &MeasureStorage{
+		path:             path,
+		favoriteFilePath: filepath.Join(path, FavoriteFileName),
+		badFilePath:      filepath.Join(path, BadFileName),
+	}
+}
+
+func (s *MeasureStorage) Load() error {
+	isExists := util.Exists(s.favoriteFilePath)
+	if !isExists {
+		err := util.WriteJSON(s.favoriteFilePath, map[string]*FavoriteNode{})
+		if err != nil {
+			return err
+		}
+	}
+	var favoriteData map[string]*FavoriteNode
+	err := util.ReadJSON(s.favoriteFilePath, &favoriteData)
+	if err != nil {
+		return err
```

Comment:
Error here means json is invalid. Should we remove the file and re-create? Same for bad node